boilerplate/flyte/precommit/hooks/pre-push

```diff
@@ -0,0 +1,41 @@
+DUMMY_SHA=0000000000000000000000000000000000000000
+
+echo "Running pre-push check; to skip this step use 'push --no-verify'"
+
+while read LOCAL_REF LOCAL_SHA REMOTE_REF REMOTE_SHA
+do
+  if [ "$LOCAL_SHA" = $DUMMY_SHA ]
+  then
+    # Branch deleted. Do nothing.
+    exit 0
+  else
+    if [ "$REMOTE_SHA" = $DUMMY_SHA ]
+    then
+      # New branch. Verify the last commit, since this is very likely where the new code is
+      # (though there is no way to know for sure). In the extremely uncommon case in which someone
+      # pushes more than 1 new commit to a branch, CI will enforce full checking.
+      RANGE="$LOCAL_SHA~1..$LOCAL_SHA"
+    else
+      # Updating branch. Verify new commits.
+      RANGE="$REMOTE_SHA..$LOCAL_SHA"
+    fi
+
+    # Verify DCO signoff. We do this before the format checker, since it has
+    # some probability of failing spuriously, while this check never should.
+    #
+    # In general, we can't assume that the commits are signed off by author
+    # pushing, so we settle for just checking that there is a signoff at all.
+    SIGNED_OFF=$(git rev-list --no-merges --grep "^Signed-off-by: " "$RANGE")
+    NOT_SIGNED_OFF=$(git rev-list --no-merges "$RANGE" | grep -Fxv "$SIGNED_OFF")
```

Comment:
This is more of relaxing the check and doing this through by grepping the sign off by string  instead of git doing the check through GPG key of the author . I would prefer to have git do the checks instead of this workaround.
@EngHabu  / @kumare3  what do you think. I agree that its one time inconvenience of setting up the GPG keys but that can be added as part of the CONTRIBUTING guide.

Reply:
so GPG is different from DCO (which is pretty annoying if you ask me)... Even if you sign your commits with GPG, you still need to `git commit -s` to please DCO... all it adds is literally that string at the end of the commit... pretty useless if you ask me. I would have much preferred to enforce GPG...